src/alpaka/AlpakaCore/initialise.h

```diff
@@ -1,28 +1,14 @@
 #ifndef AlpakaCore_initialise_h
 #define AlpakaCore_initialise_h
 
-#include "AlpakaCore/alpakaConfig.h"
+#include "AlpakaCore/config.h"
 
 namespace cms::alpakatools {
 
   // note: this function is not thread-safe
   template <typename TPlatform>
   void initialise(bool verbose = false);
 
-  // explicit template instantiation declaration
-#ifdef ALPAKA_ACC_CPU_B_SEQ_T_SEQ_PRESENT
-  extern template void initialise<alpaka_serial_sync::Platform>(bool);
-#endif
-#ifdef ALPAKA_ACC_CPU_B_TBB_T_SEQ_PRESENT
-  extern template void initialise<alpaka_tbb_async::Platform>(bool);
-#endif
-#ifdef ALPAKA_ACC_GPU_CUDA_PRESENT
-  extern template void initialise<alpaka_cuda_async::Platform>(bool);
-#endif
-#ifdef ALPAKA_ACC_GPU_HIP_PRESENT
-  extern template void initialise<alpaka_rocm_async::Platform>(bool);
-#endif
-
```

Comment:
@makortel I'm not sure about this change, what do you think ?

Reply:
I think it make sense to treat `intialise.h` and `device.h` consistently. While the "implicit instantiation" seems to be sufficient for these function templates to work, the `extern template` would have the benefit that it would prevent implicit instantiations. From https://en.cppreference.com/w/cpp/language/function_template
> An explicit instantiation declaration (an extern template) prevents implicit instantiations: the code that would otherwise cause an implicit instantiation has to use the explicit instantiation definition provided somewhere else in the program.

Maybe the `extern template` would make the instantiations of a wrong Platform to lead a compilation failure (rather than a link failure)?